internal/fs/read_cache_test.go

```diff
@@ -693,10 +694,66 @@ func (t *FileCacheWithDefaultCacheLocation) SetUpTestSuite() {
 
 func (t *FileCacheWithDefaultCacheLocation) TearDown() {
 	t.fsTest.TearDown()
-	err := os.RemoveAll(CacheLocation)
+	err := os.RemoveAll(path.Join(os.TempDir(), util.FileCache))
 	AssertEq(nil, err)
 }
 
 func (t *FileCacheWithDefaultCacheLocation) DefaultLocationIsTempDir() {
-	sequentialReadShouldPopulateCache(&t.fsTest, os.TempDir())
+	sequentialReadShouldPopulateCache(&t.fsTest, path.Join(os.TempDir(), util.FileCache))
+}
+
+// Test to check cache is deleted at the time pf unmounting.
+type FileCacheDestroyTest struct {
+	fsTest
+}
+
+func init() {
+	RegisterTestSuite(&FileCacheDestroyTest{})
+}
+
+func (t *FileCacheDestroyTest) SetUpTestSuite() {
+	t.serverCfg.ImplicitDirectories = true
+	t.serverCfg.MountConfig = &config.MountConfig{
+		FileCacheConfig: config.FileCacheConfig{
+			MaxSizeInMB:               -1,
+			DownloadFileForRandomRead: true,
+		},
+		CacheLocation: config.CacheLocation(CacheLocation),
+	}
+	t.serverCfg.AllowOther = true
+	t.fsTest.SetUpTestSuite()
+}
+
+func (t *FileCacheDestroyTest) TearDownTestSuite() {
+	// do nothing as fs is unmounted in the test itself
+}
+
+func (t *FileCacheDestroyTest) TearDown() {
```

Comment:
I am not sure, if we need TearDown/TearDownTestSuite here. If not can we remove this?

Reply:
We don't need to do anything in TearDown/TestSuite that's why we have defined the function, otherwise it uses the implementation from fsTest.